extensions/redis-cache/src/main/java/io/druid/client/cache/RedisCache.java

```diff
@@ -0,0 +1,288 @@
+  {
+    final Map<byte[], NamedKey> keyLookup = Maps.uniqueIndex(
+      keys,
+      new Function<NamedKey, byte[]>()
+      {
+        @Override
+        public byte[] apply(
+          @Nullable NamedKey input
+        )
+        {
+          return computeKeyHash(prefix, input);
+        }
+      }
+    );
+    final Map<NamedKey, byte[]> results = Maps.newHashMap();
+
+    // Sometimes broker passes empty keys list to getBulk()
+    if (keyLookup.size() == 0) {
+      return results;
+    }
+
+    final List<byte[]> keysList = new ArrayList<>(keyLookup.keySet());
+    final List<byte[]> valuesList;
+
+    try (final Jedis client = pool.getResource()) {
+      valuesList = client.mget(keysList.toArray(new byte[keysList.size()][]));
+    }
+    catch (JedisConnectionException ex) {
+      if (isTimeout(ex)) {
+        timeoutCount.incrementAndGet();
+      }
+
+      throw Throwables.propagate(ex);
+    }
+    catch (JedisException ex) {
+      errorCount.incrementAndGet();
+      log.warn(ex, "Unable to get cache items");
+      return results;
+    }
+
+    final int length = valuesList.size();
+    int i = 0;
+
+    while (i < length) {
+      final byte[] value = valuesList.get(i);
+      if (value != null) {
+        final NamedKey key = keyLookup.get(keysList.get(i));
+        try {
+          final byte[] deserialized = deserialize(key, value);
+          hitCount.incrementAndGet();
+          results.put(key, deserialized);
+        }
+        catch (LZ4Exception ex) {
+          errorCount.incrementAndGet();
+          log.warn(ex, "Error decompressing cache item");
+        }
+      }
+      else {
+        missCount.incrementAndGet();
+      }
+
+      i++;
+    }
+
+    return results;
+  }
+
+  @Override
+  public void close(String namespace)
+  {
+    final byte[] pattern = redisKey(prefix, hash(namespace), "*");
+    try (final Jedis client = pool.getResource()) {
+      final Set<byte[]> keys = client.keys(pattern);
+      if (keys.size() == 0) {
+        return;
+      }
+
+      client.del(keys.toArray(new byte[keys.size()][]));
+    }
+    catch (JedisConnectionException ex) {
+      Throwables.propagate(ex);
+    }
+    catch (JedisException ex) {
+      log.warn(ex, "Error cleaning cache items");
+    }
+  }
+
+  @Override
+  public CacheStats getStats()
+  {
+    return new CacheStats(
+        hitCount.get(),
+        missCount.get(),
+        0,
+        0,
+        0,
+        timeoutCount.get(),
+        errorCount.get()
+    );
+  }
+
+  @Override
+  public boolean isLocal()
+  {
+    return false;
+  }
+
+  @Override
+  public void doMonitor(ServiceEmitter emitter)
+  {
+    //TODO
+  }
+
+  private static boolean isTimeout(JedisConnectionException ex)
+  {
+    //FIXME NoSuchElementException is thrown not only in case of timeout
+    return ex.getCause() instanceof NoSuchElementException;
+  }
+
+  private byte[] deserialize(NamedKey key, byte[] bytes)
```

Comment:
same here, we might want to move some of the common memcached / redis cache code to druid-processing and reuse it.